src/internet_identity/src/stats/event_stats/tests.rs

```diff
@@ -637,6 +637,60 @@ fn should_add_prune_events_to_prune_aggregations() {
     assert_event_count_consistent(&mut storage);
 }
 
+#[test]
+fn should_prune_zero_weighted_events() {
+    let mut storage = test_storage();
+    let ii_domain = Some(IIDomain::Ic0App);
+    let event = EventData {
+        event: Event::PrepareDelegation(PrepareDelegationEvent {
+            ii_domain: ii_domain.clone(),
+            frontend: EXAMPLE_URL.to_string(),
+            session_duration_ns: to_ns(0),
+        }),
+    };
+
+    // no events initially
+    assert_eq!(storage.event_data.len(), 0);
+
+    // add multiple zero weighted event
+    update_events_internal(event.clone(), TIMESTAMP, &mut storage);
+    update_events_internal(event.clone(), TIMESTAMP, &mut storage);
```

Comment:
Does it make sense to add a zero event from a different frontend (and therefore a different key)?

Reply:
Yes, done.